analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/LocksReleasedAllPathsBase.cs

```diff
@@ -179,12 +203,72 @@ private ProgramState RemoveLock(SymbolicContext context, ISymbol symbol)
         private static ProgramState AddLock(SymbolicContext context, IOperation operation) =>
             context.State.SetOperationConstraint(operation, context.SymbolicValueCounter, LockConstraint.Held);
 
-        private static ISymbol FirstArgumentSymbol(IInvocationOperationWrapper invocation) =>
-            IArgumentOperationWrapper.FromOperation(invocation.Arguments.First()).Value.TrackedSymbol();
+        private static ISymbol ArgumentSymbol(IInvocationOperationWrapper invocation, int argumentIndex, string parameterName) =>
+            invocation.Arguments[argumentIndex].ToArgument() is var argument
+            && argument.Parameter.Name == parameterName
+                ? argument.Value.TrackedSymbol()
+                : invocation.Arguments.SingleOrDefault(x => x.ToArgument() is var namedArgument && namedArgument.Parameter.Name == parameterName).ToArgument().Value.TrackedSymbol();
+
+        private static ISymbol FindLockSymbolIfConditionalReturnValue(SymbolicContext context)
+        {
+            if (context.Operation.Instance.AsInvocation() is not { } invocation
```

Comment:
This will be easier to follow with positive condition and nested if.
```
if(... is invocation && invocation....Is(...Boolean))
{
   if(..)
   elseif(..)
}
return null;    - here it completely makes sense without `else`, highlighting the fact that there are other paths somewhere in the middle ending up here
```